### PR TITLE
PT-333/Support for calendars

### PR DESCRIPTION
**Pull request by florianmski**
> Tracked in JIRA by [PT-333](https://novoda.atlassian.net/browse/PT-333)

### Scope of the PR

Add a dashboard that show what is the next calendar event of a specific calendar.
We want to have more granular dashboards so we have a basic set of dashboards when we're going to have specific dashboards client for projects / office location.

webdashboard

### CODE REVIEW

**Comment by xrigau** [comment]
stalky calendar is stalky

File: node-server/config.json.sample [hunk 1]
```diff
@@ -13,6 +13,14 @@
       "email": "",
       "password": "",
       "exclude": []
+    },
+    "google": {
+    "private_key": "",
```
Comment (frapontillo) [inline node-server/config.json.sample]: Where has my ~~country~~ indentation gone

File: node-server/server/widget/calendar.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      return Promise.resolve({
+        widgetKey: 'calendar',
+        template: 'classic',
+        payload: {
+        	title: `📅<br/>Next event @${startTime}`,
+        	text: event.summary
+        }
+      });
+    } else {
+      return Promise.reject("No more events for today");
```
Comment (frapontillo) [inline node-server/server/widget/calendar.js]: Just throwing an error causes the `catch` block to be called, but this is fine too.

File: node-server/server/widget/calendar.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        }
+        resolve(response);
+      });
+    });
+  }
+}
+
+function toRuleResult(data) {
+    if (data.items.length > 0) {
+      var event = data.items[0];
```
Comment (frapontillo) [inline node-server/server/widget/calendar.js]: `var`? 💣

**Comment by takecare** [comment]
nice 👍

File: node-server/config.json.sample (hunk 1 above)
Comment (florianmski) [inline node-server/config.json.sample]: Sorry, trying to use VIM, didn't get to the indenting chapter yet 😞

File: node-server/server/widget/calendar.js (hunk 2 above)
Comment (florianmski) [inline node-server/server/widget/calendar.js]: This seems to be what's used in the other js files, following the convention 👍

File: node-server/server/widget/calendar.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+          return reject(err);
+        }
+        resolve(response);
+      });
+    });
+  }
+}
+
+function toRuleResult(data) {
+  let filteredEvents = data.items.filter(event => keepEvent(event));
```
Comment (frapontillo) [inline node-server/server/widget/calendar.js]: So here you can just pass the keepEvent function as a reference. Not needed in this pr, just laying it out.

Comment (florianmski) [inline node-server/server/widget/calendar.js]: Good to know 👍

**Comment by xrigau** [comment]
Very cool @florianmski ! @frapontillo good to :shipit: ?

**Comment by frapontillo** [comment]
Waiting for the branch update to pass, then good to 🚢

**Comment by xrigau** [comment]
Time to 🚀

**Pull request merged**